site/frontend/src/pages/compare/runtime/common.ts

```diff
@@ -25,7 +25,8 @@ export function computeRuntimeComparisonsWithNonRelevant(
   function shouldShowTestCase(comparison: TestCaseComparison<RuntimeTestCase>) {
     const name = comparison.testCase.benchmark;
     const nameFilter = filter.name && filter.name.trim();
-    return !nameFilter || name.includes(nameFilter);
+    const nameFiltered = !nameFilter || name.match(new RegExp(nameFilter));
```

Comment:
This logic will be slightly more complicated (see the other comment), so it would be nice to unify the logic in some shared function, something like `filterBenchmarkName(benchmarkName, filterName)` and put it e.g. in `site/src/pages/compare/shared.ts`.